src/pip/_internal/models/format_control.py

```diff
@@ -26,10 +26,10 @@ def __init__(self, no_binary=None, only_binary=None):
     def __eq__(self, other):
         # type: (object) -> bool
         if not isinstance(other, self.__class__):
-            return NotImplemented
+            return False
```

Comment:
I think it’s better to return `NotImplemented` here. It would trigger Python to raise a more readable exception saying the two types are not comparable.

p.s. `False` is correct return value IMO for the next if branch.

Reply:
Done